src/store/document-drive.ts

```diff
@@ -43,74 +48,237 @@ function debounceOperations(
     };
 }
 
+type FileNodeDocument =
+    | {
+          driveId: string;
+          documentId: string;
+          documentType: string;
+          document: Document | undefined;
+          status: 'LOADING' | 'ERROR';
+      }
+    | {
+          driveId: string;
+          documentId: string;
+          documentType: string;
+          document: Document;
+          status: 'LOADED';
+      }
+    | undefined;
+
+const documentCacheAtom = atom(new Map<string, Document>());
+
+const singletonFileNodeDocumentAtom = atom<FileNodeDocument>(undefined);
```

Comment:
Since we are managing the selected document in hooks, it was causing parallel requests to be made, one for each document using the hook.
I've added these atoms which prevent that and use a single request.